lib/private/EventDispatcher/EventDispatcher.php

```diff
@@ -54,6 +55,12 @@ public function __construct(SymfonyDispatcher $dispatcher,
 		$this->dispatcher = $dispatcher;
 		$this->container = $container;
 		$this->logger = $logger;
+
+		// inject the event dispatcher into the logger
+		// this is done here because there is a cyclic dependency between the event dispatcher and logger
+		if ($this->logger instanceof Log or $this->logger instanceof Log\PsrLoggerAdapter) {
```

Comment:
```suggestion
		if ($this->logger instanceof Log || $this->logger instanceof Log\PsrLoggerAdapter) {
```
This is a backport so it should be left as-is, but we usually use `||`/`&&` and not `or`/`and`